pkg/model/form.go

```diff
@@ -0,0 +1,56 @@
+package model
+
+import (
+	"fmt"
+	"gopkg.in/mgo.v2/bson"
+	"time"
+)
+
+type FormWidget struct {
+	ID        int64       `json:"id" bson:"_id"`
+	Type      string      `json:"type" bson:"type"`
+	Component string      `json:"component" bson:"component"`
+	Title     string      `json:"title" bson:"title"`
+	Wrapper   interface{} `json:"wrapper" bson:"wrapper"`
+	Props     interface{} `json:"props" bson:"props"`
+}
+
+type FormStep struct {
+	ID      int64        `json:"id" bson:"_id"`
+	Name    string       `json:"name" bson:"name"`
+	Widgets []FormWidget `json:"widgets" bson:"widgets"`
+}
+
+type Form struct {
+	ID             bson.ObjectId `json:"id" bson:"_id"`
+	Status         string        `json:"status" bson:"status"`
+	Settings       interface{}   `json:"settings" bson:"settings"`
+	Header         interface{}   `json:"header" bson:"header"`
+	Footer         interface{}   `json:"footer" bson:"footer"`
+	FinishedScreen interface{}   `json:"finishedScreen" bson:"finishedScreen"`
+	Steps          []FormStep    `json:"steps" bson:"steps"`
+	CreatedBy      interface{}   `json:"created_by" bson:"created_by"` // Todo, decide how to represent ownership here
+	UpdatedBy      interface{}   `json:"updated_by" bson:"updated_by"` // Todo, decide how to represent ownership here
```

Comment:
Do we do soft delete on the forms or just delete the whole form? It would be good to do a soft delete and "archive" the forms to get later historical data on forms.